spec/lib/rufo/formatter_source_specs/2.3/squiggly_heredoc.rb.spec

```diff
@@ -272,4 +272,36 @@ EOF
   #{2}
 EOF
 
+#~# ORIGINAL
+if true
+  <<~EOF
+  EOF
+end
+
+#~# EXPECTED
+
+#~# ORIGINAL
+begin
+  <<~EOF
+  EOF
+
+  true
+end
 
+#~# EXPECTED
+
+#~# ORIGINAL
+-> {
+  x().x <<~EOF
+  EOF
+}
+
+#~# EXPECTED
+
+#~# ORIGINAL
+x do
+  x(<<-EOF).x
```

Comment:
Every time I see code like this I'm surprised that it's valid Ruby

Reply:
Agreed, after working on rufo I want to campaign for removing heredocs 🤣